srml/slashing/src/lib.rs

```diff
@@ -0,0 +1,108 @@
+#![deny(missing_docs, rust_2018_idioms)]
+
+//! Abstract slashing interface
+//!
+//! That gives functionality to specialize slashing and misconduct for a given type
+//! In order to customize severity level and misconduct fees.
+//!
+//! TODO(niklasad1): provide default impl?
+//! TODO(niklasad1): move the `srml/support/src/traits.rs`?
+//! TODO(niklasad1): chain reversion are ignored, do we need specific method/function for that?
```

Comment:
what exactly do you mean chain reversion?

Reply:
>  In order to slash somebody, we want to have an “attestation of misconduct” that is objective, short, and valid on all forks. Moreover it should remain valid in case of chain reversion. We also need to ensure that two attestations for the same misconduct cannot both be valid simultaneously, so that we don’t punish twice for the same crime

Oops, I think I just realized that this should be probably be handled at another level (such as the caller in this context). Ok?

Reply:
yup, that's handled by whatever actually checks proofs of misbehavior